### ISSUE TITLE
[Submission]: qiskit-classroom-converter

### ISSUE THREAD

**Issue by minwook-shin** [qiskit-community/ecosystem#462]
### Github repo

https://github.com/KMU-quantum-classroom/qiskit-classroom-converter

### Description

Convert quantum circuits, matrices, and bra-ket strings.

This converter includes the following conversion functions: quantum circuit to bra-ket notation, quantum circuit to matrix, matrix to quantum circuit, bra-ket notation to matrix

### License

Apache License 2.0

### Tags

converter

### Website

https://github.com/KMU-quantum-classroom

**Comment by mickahell** [comment]
It seems you have some problem with your requirements :
```
2023-07-28 06:48:21 ecosystem INFO Getting requirements to build editable: finished with status 'error'
2023-07-28 06:48:21 ecosystem INFO error: subprocess-exited-with-error
...
2023-07-28 06:48:21 ecosystem INFO × Getting requirements to build editable did not run successfully.
2023-07-28 06:48:21 ecosystem INFO │ exit code: 1
```

**Comment by mickahell** [comment]
We currently have an evolve waiting to tests in order to not require tests with latest and dev release of Qiskit but only with your standard tests.

I almost finished it, just need to fully test it. I think it gonna be merged in the next week.

**Comment by frankharkins** [comment]
@minwook-shin thanks for your submission! Unfortunately the description can't handle markdown; can you edit it to just be plain text?

**Comment by minwook-shin** [comment]
@frankharkins changed it to plain text right now! :)

**Comment by minwook-shin** [comment]
https://github.com/KMU-quantum-classroom/qiskit-classroom-converter/releases/tag/0.0.2 
and released version 0.0.2, which is update to qiskit 0.44.0.

**Comment by mickahell** [comment]
@minwook-shin it seems you still have the same requirement problem and you have it at every check. Even the standard one who doesn't modify any import.

**Comment by minwook-shin** [comment]
@mickahell 
removed duplicate package name by extra_dependencies in ecosystem.json.
https://github.com/KMU-quantum-classroom/qiskit-classroom-converter/blob/ebd114c47b051c8c507dc030be88bac0c4f35b79/ecosystem.json#L6
do i understand that correctly?

**Comment by mickahell** [comment]
> @mickahell 
> 
> removed duplicate package name by extra_dependencies in ecosystem.json.
> 
> https://github.com/KMU-quantum-classroom/qiskit-classroom-converter/blob/ebd114c47b051c8c507dc030be88bac0c4f35b79/ecosystem.json#L6
> 
> do i understand that correctly?

You can delete the field if not used. Anyway you still have the error we need to investigate more to see where is the problem

**Comment by minwook-shin** [comment]
@mickahell for your information, i used pyproject.toml instead setup.py, and versioning setuptools>=68.0.

**Comment by mickahell** [comment]
@minwook-shin I maybe find something, could you add that : `py-modules = []` in your `pyproject.toml [tool.setuptools]` please.

ref : https://github.com/pypa/setuptools/issues/3197

**Comment by minwook-shin** [comment]
@mickahell @frankharkins I have been looking for that issue.
installing package that use setuptools of pyproject.toml in isolated container (like github action) seems to cause issues. 

Issue ref : https://github.com/pypa/setuptools/issues/3197#issuecomment-1086866808
Temporary Solution ref : https://github.com/numpy/numpy/issues/21288#issuecomment-1086875871

**Comment by minwook-shin** [comment]
@mickahell the method is labeled as beta, so it seems risky to add right now.
So we're testing it before deploy this way.
I will tell you when apply it.
```
_BetaConfiguration: Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.
```

**Comment by mickahell** [comment]
> @mickahell the method is labeled as beta, so it seems risky to add right now. So we're testing it before deploy this way. I will tell you when apply it.
> 
> ```
> _BetaConfiguration: Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.
> ```

Oh sorry I didn't saw that :o my bad



> @mickahell @frankharkins I have been looking for that issue. installing package that use setuptools of pyproject.toml in isolated container (like github action) seems to cause issues.
> 
> Issue ref : [pypa/setuptools#3197 (comment)](https://github.com/pypa/setuptools/issues/3197#issuecomment-1086866808) Temporary Solution ref : [numpy/numpy#21288 (comment)](https://github.com/numpy/numpy/issues/21288#issuecomment-1086875871)

I don't really think that the problem.
Indeed, when you are running your tox in your Actions everything seems to works fine by running the cmd `tox` alone. Us, we are also executing tox in a subsystem env but with the cmd `tox -epy39`

**Comment by minwook-shin** [comment]
@mickahell 
i'm late in checking the package works on Test PyPI.
and we have released version 0.0.3, which fixe the setuptools package code.
Please run the build system.
https://github.com/KMU-quantum-classroom/qiskit-classroom-converter/releases/tag/0.0.3

**Comment by frankharkins** [comment]
Looks good! Thanks @mickahell @minwook-shin 🎉
